agent/proxycfg/snapshot.go

```diff
@@ -379,6 +379,29 @@ func (c *configSnapshotIngressGateway) IsEmpty() bool {
 		len(c.WatchedUpstreamEndpoints) == 0
 }
 
+// ValidUpstreams returns the list of upstreams that have enough data to be used.
+func (c *configSnapshotIngressGateway) ValidUpstreams() structs.Upstreams {
```

Comment:
curious if this should be called something like ~ `ReadyUpstreams` ? Or `UpstreamsWithData`

Reply:
I've used a consistent naming with [cfgSnap.TerminatingGateway.ValidServices()](https://github.com/hashicorp/consul/blob/a38be58eaa029f7383b78de9b23f9a5dbe2f7054/agent/xds/routes.go#L90). That is what @rboyer suggested in [his issue](https://github.com/hashicorp/consul/issues/11291).

I agree that `ReadyUpstreams` might be more appropriate.

Let me know if we prefer consistency or better naming.

Reply:
I cast my vote for `ReadyUpstreams` as it seems more descriptive here